src/main/java/org/jahia/modules/visibility/conditions/DayOfWeekVisibilityCondition.java

```diff
@@ -0,0 +1,74 @@
+package org.jahia.modules.visibility.conditions;
+
+import org.jahia.registries.ServicesRegistry;
+import org.jahia.services.content.JCRNodeWrapper;
+import org.jahia.services.visibility.BaseVisibilityConditionRule;
+import org.jahia.utils.DateUtils;
+import org.jahia.utils.i18n.Messages;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.jcr.PathNotFoundException;
+import javax.jcr.RepositoryException;
+import javax.jcr.Value;
+import java.util.Calendar;
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Set;
+
+/**
+ * Defines the visibility condition, based on the day of the week. The content is considered visible if the current day of the week is among
+ * the selected days.
+ *
+ * @author Sergiy Shyrkov
+ */
+public class DayOfWeekVisibilityCondition extends BaseVisibilityConditionRule {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DayOfWeekVisibilityCondition.class);
+
+    public String getGWTDisplayTemplate(Locale locale) {
+        return Messages.get(ServicesRegistry.getInstance().getJahiaTemplateManagerService().getTemplatePackage("Jahia Visibility"), "label.dayOfWeekCondition.xtemplate", locale);
+    }
+
+    public boolean matches(JCRNodeWrapper node) {
+        boolean visible = true;
+        Set<Integer> days = null;
+        try {
+            Value[] values = node.getProperty("dayOfWeek").getValues();
+            days = new HashSet<Integer>(7);
+            for (Value value : values) {
+                Integer dow = DateUtils.getDayOfWeek(value.getString());
+                if (dow != null) {
+                    days.add(dow);
+                } else {
+                    LOGGER.warn("Unknown day of week '{}' in a visibility condition for node {}. Skipping.", value.getString(), node.getPath());
+                }
+            }
+        } catch (PathNotFoundException e) {
+            if (LOGGER.isDebugEnabled()) {
```

Comment:
There is some of LOGGER.isDebugEnabled() is the code, does it make sense to keep them?